gfx/video_thread_wrapper.c

```diff
@@ -617,11 +615,13 @@ static bool video_thread_frame(void *data, const void *frame_,
 
       scond_signal(thr->cond_thread);
 
-#if defined(HAVE_MENU)
+#ifdef HAVE_MENU
       if (thr->texture.enable)
       {
-         while (thr->frame.updated)
+         do
```

Comment:
Was this a bug in the original code? i.e. do we need an `scond_wait()` even if `thr->frame.updated` is false...?

(I am not very familiar with this code, so I'm not sure of the expected behaviour!)

Reply:
It will always be true at that point because we've just set it to true (and we've the lock acquired). I've changed it to a "do while" instead exactly for this reason.

I am not 100% sure why it needs to block until the driver thread has processed the frame if you've menus, but I trust the author knew better.